data/qsort_with_docs.c

```
@@ -0,0 +1,64 @@
+            i++;
+            
+            swap(&arr[i], &arr[j]);
+        }
+    }
+    
+    swap(&arr[i + 1], &arr[high]);
+    return i + 1; 
+}
+
+
+void quickSort(int arr[], int low, int high)
```

Clang (14.0.6)

[warning] function 'quickSort' is within a recursive call chain

(misc-no-recursion)